modules/p2/ting_carousel/ting_carousel.module

```diff
@@ -67,19 +67,19 @@ function ting_carousel_preprocess_ting_object(&$variables, $hook) {
     return;
   }
 
-  // Because of browser compatibility we truncate server time.
-  $variables['content']['ting_title'][0]['#markup'] = truncate_utf8($variables['content']['ting_title'][0]['#markup'], 33, FALSE, TRUE);
-
   $entity_uri = entity_uri('ting_object', $variables['object']);
   if (isset($entity_uri)) {
     $entity_link = url($entity_uri['path'], array('absolute' => TRUE));
-    $variables['content']['ting_cover']['#prefix'] = '<div class="contentbox"><a href="' . $entity_link . '">';
-    $variables['content']['ting_title']['#suffix'] = '</a></div>';
+    $variables['content']['ting_cover']['#prefix'] = '<div class="contentbox">';
+    $variables['content']['group_overlay']['ting_title']['#prefix'] = '<a href="' . $entity_link . '" class="item-link">';
+    $variables['content']['group_overlay']['ting_title']['#suffix'] = '</a>';
+    $variables['content']['group_overlay']['#suffix'] = '</div></div>';
```

Comment:
I cannot follow the prefix/suffix additions here.

`ting_cover` adds one `<div>` as a prefix. `group_overlay` adds two `</div>` as a suffix. That does not add up to me.

If this is intentional then please add code comments explaining the logic.